samples/tab-approval/csharp/TabRequestApproval/Controllers/HomeController.cs

```diff
@@ -0,0 +1,200 @@
+            {
+                ViewBag.Message = "No record found";
+            }
+            else
+            {
+                var request = currentTaskList.FirstOrDefault(p => p.taskId.ToString() == taskId);
+                ViewBag.TaskDetails = request;
+            }
+            return View("Request");
+        }
+
+        // Redirect to auth page.
+        [Route("TabAuth")]
+        public ActionResult Auth()
+        {
+            return View("TabAuth");
+        }
+
+        // Send notification to manager about request.
+        [HttpPost]
+        [Route("SendNotificationToManager")]
+        public async Task<ActionResult> SendNotificationToManager(RequestInfo taskInfo)
+        {
+            try
+            {
+                var currentTaskList = new List<RequestInfo>();
+                List<RequestInfo> taskList = new List<RequestInfo>();
+                _taskList.TryGetValue("taskList", out currentTaskList);
+
+                var request = taskInfo;
+                request.taskId = Guid.NewGuid();
+                request.status = "Pending";
+
+                if (currentTaskList == null)
+                {
+                    taskList.Add(request);
+                    _taskList.AddOrUpdate("taskList", taskList, (key, newValue) => taskList);
+                    ViewBag.TaskList = taskList;
+                }
+                else
+                {
+                    currentTaskList.Add(request);
+                    _taskList.AddOrUpdate("taskList", currentTaskList, (key, newValue) => currentTaskList);
+                    ViewBag.TaskList = currentTaskList;
+                }
+
+                var graphClient = SimpleGraphClient.GetGraphClient(taskInfo.access_token);
+                var graphClientApp = SimpleGraphClient.GetGraphClientforApp(_configuration["AzureAd:MicrosoftAppId"], _configuration["AzureAd:MicrosoftAppPassword"], _configuration["AzureAd:TenantId"]);
+                var user = await graphClient.Users[taskInfo.managerName]
+                          .Request()
+                          .GetAsync();
+
+                var installedApps = await graphClient.Users[user.Id].Teamwork.InstalledApps
+                                   .Request()
+                                   .Expand("teamsApp")
+                                   .GetAsync();
+
+                var installationId = installedApps.Where(id => id.TeamsApp.DisplayName == "Tab Approval").Select(x => x.TeamsApp.Id);
+                var userName = user.UserPrincipalName;
+   
+                var url = "https://teams.microsoft.com/l/entity/"+installationId.ToList()[0]+"/request?context={\"subEntityId\":\""+ request.taskId+"\"}";
+                var topic = new TeamworkActivityTopic
+                {
+                    Source = TeamworkActivityTopicSource.Text,
+                    Value = $"{taskInfo.title}",
+                    WebUrl = url
+                };
+
+                var activityType = "approvalRequired";
```

Comment:
You can directly pass the value rather than creating a variable.

Reply:
removed